src/main/java/org/apache/hadoop/util/LineReader.java

```diff
@@ -244,6 +244,10 @@ private int readDefaultLine(Text str, int maxLineLength, int maxBytesToConsume)
             int appendLength = readLength - newlineLength;
             if (appendLength > maxLineLength - txtLength) {
                 appendLength = maxLineLength - txtLength;
+                if (appendLength > 0) {
+                    // We want to fail the query when the line length is over the limit.
```

Comment:
This comment is out of place here, since this is a generic library. We could say *"fail the read"* instead. The same goes for the other comments.